src/odemis/driver/test/orsay_test.py

```diff
@@ -0,0 +1,560 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on 6 April 2021
+
+@author: Arthur Helsloot
+
+Copyright © 2021-2023 Arthur Helsloot, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+
+from __future__ import division
+
+import os
+import unittest
+from time import sleep
+from odemis.driver import orsay
+
+TEST_NOHW = (os.environ.get("TEST_NOHW", 0) != 0)  # Default to Hw testing
+
+CONFIG_PSUS = {"name": "pneumatic-suspension", "role": "pneumatic-suspension"}
+CONFIG_PRESSURE = {"name": "pressure", "role": "chamber"}
+CONFIG_PSYS = {"name": "pumping-system", "role": "pumping-system"}
+CONFIG_UPS = {"name": "ups", "role": "ups"}
+
+CONFIG_ORSAY = {"name": "Orsay", "role": "orsay", "host": "192.168.56.101",
+                "children": {"pneumatic-suspension": CONFIG_PSUS,
+                             "pressure": CONFIG_PRESSURE,
+                             "pumping-system": CONFIG_PSYS,
+                             "ups": CONFIG_UPS}
+                }
+
+
+class TestOrsayStatic(unittest.TestCase):
+    """
+    Tests which don't need an Orsay component ready
+    """
+
+    def test_creation(self):
+        """
+        Test to create an Orsay component
+        """
+        try:
+            oserver = orsay.OrsayComponent(**CONFIG_ORSAY)
+        except Exception as e:
+            self.fail(e)
+        self.assertEqual(len(oserver.children.value), 4)
+
+        oserver.terminate()
+
+
+class TestOrsay(unittest.TestCase):
+    """
+    Tests to run on the main Orsay component
+    """
+
+    def setUp(self):
```

Comment:
This start/stop the components before every test cases. That works, but might be quite slow. It'd usually nicer to use 
 ```
@classmethod
def setUpClass(cls):
```
(and tearDownClass)